includes/helpers/class-listing-image.php

```diff
@@ -125,8 +125,9 @@ public static function maybe_delete_image( $id, $listing_id ) {
 
 		// If the current listing is still cached, don't count it.
 		$linked_listings = array_diff( $linked_listings, array( $listing_id ) );
-
-		if ( empty( $linked_listings ) ) {
+		$linked_posts    = self::get_linked_posts( $id );
```

Comment:
Let's only check for linked posts if there are no linked listings to cut down on DB calls. Also, it would be great to move some of this to a new function that returns true or false if there is another post or listing using the image.

Reply:
Changes applied. I have separated the check for listings and posts to a separate private function `check_if_attachment_is_linked` that returns a boolean and this is called once in the function `maybe_delete_image`